src/vs/workbench/contrib/terminal/common/terminalExtensionPoints.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as extensionsRegistry from 'vs/workbench/services/extensions/common/extensionsRegistry';
+import { ITerminalTypeContribution, ITerminalContributions, terminalContributionsDescriptor } from 'vs/workbench/contrib/terminal/common/terminal';
+import { flatten } from 'vs/base/common/arrays';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+
+// terminal extension point
+export const terminalsExtPoint = extensionsRegistry.ExtensionsRegistry.registerExtensionPoint<ITerminalContributions>(terminalContributionsDescriptor);
+
+export interface ITerminalContributionService {
+	readonly _serviceBrand: undefined;
+
+	readonly terminalTypes: ReadonlyArray<ITerminalTypeContribution>;
+}
+
+export const ITerminalContributionService = createDecorator<ITerminalContributionService>('terminalContributionsService');
```

Comment:
I don't see any other `IXContributionService`s, what's the usual pattern for this sort of thing?

Reply:
They seem to usually be gathered up in some contribution service, e.g. https://github.com/microsoft/vscode/blob/master/src/vs/workbench/contrib/codeActions/common/codeActionsContribution.ts#L54 or https://github.com/microsoft/vscode/blob/master/src/vs/workbench/contrib/tasks/common/taskDefinitionRegistry.ts#L88

Often the contribution services do additional things with the contribution information, but here there's no need for that yet.

Reply:
Let's stick to this for now I guess, should be easy to change later as needed.